src/test/spec/unified_runner/test_file.rs

```diff
@@ -133,6 +134,11 @@ impl RunOnRequirement {
                 return false;
             }
         }
+        if let Some(csfle) = &self.csfle {
+            if *csfle && std::env::var("KMS_PROVIDERS").is_err() {
```

Comment:
if `csfle` is set in the test file, does that mean that `KMS_PROVIDERS` should be set? or is it valid for one of these values to be set but not the other? if it's the former, perhaps we should panic or log some kind of message here to avoid skipping CSFLE tests silently